### PR TITLE
Max idle conns per host

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #26519

Hopefully, this will at least somehow help with #26519 

@gmarek @krousey @lavalamp

### CODE REVIEW

**Comment by wojtek-t** [comment]
@krousey - friendly ping

**Comment by gmarek** [comment]
Oh, I forgot to mention that LGTM to me, but I'd have @krousey give a final one.

File: test/e2e/service.go [hunk 1]
```diff
@@ -69,9 +69,7 @@ var _ = framework.KubeDescribe("Services", func() {
 	var c *client.Client
 
 	BeforeEach(func() {
-		var err error
-		c, err = framework.LoadClient()
-		Expect(err).NotTo(HaveOccurred())
+		c = f.Client
```
Comment (krousey) [inline test/e2e/service.go]: Does this really need to run before each? Can't it just be assigned above?

File: test/e2e/petset.go [hunk 2]
```diff
@@ -62,9 +62,7 @@ var _ = framework.KubeDescribe("PetSet", func() {
 		// dynamic volume provisioner.
 		framework.SkipUnlessProviderIs("gce")
 
-		var err error
-		c, err = framework.LoadClient()
-		Expect(err).NotTo(HaveOccurred())
+		c = f.Client
```
Comment (krousey) [inline test/e2e/petset.go]: Can't this just be assigned above?

**Comment by krousey** [comment]
LGTM with a couple of minor comments.

File: test/e2e/petset.go (hunk 2 above)
Comment (wojtek-t) [inline test/e2e/petset.go]: This seems to be a common pattern in tests to assing client in BeforeEach. I wanted to be consistent with other places, e.g.:
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/daemon_set.go#L85
https://github.com/kubernetes/kubernetes/blob/master/test/e2e/kubectl.go#L179
...
(there are 19 such places)

If you think we should change that, can I send a separate PR to fix all of those in one, and for this one leave as is?

Comment (krousey) [inline test/e2e/petset.go]: Yea. I didn't realize it was that wide spread.

**Pull request merged**